kubeflow/kubeflux/all.libsonnet

```diff
@@ -0,0 +1,283 @@
+            "namespace": namespace
+          }
+        ]
+      },
+
+    secret:: {
+      // K8s Deployment,
+      "apiVersion": "v1",
+      "kind": "Secret",
+      "metadata": {
+        "name": "flux-git-deploy",
+      },
+      "type": "Opaque"
+    },
+
+    flux:: {
+      // K8s Deployment,
+      "apiVersion": "apps/v1beta1",
+      "kind": "Deployment",
+      "metadata": {
+        "name": "flux",
+      },
+      "spec": {
+        "replicas": 1,
+        "strategy": {
+          "type": "Recreate"
+        },
+        "template": {
+          "metadata": {
+            "labels": {
+              "name": "flux"
+            }
+          },
+          "spec": {
+            "containers": [
+              {
+                "args": [
+                  "--ssh-keygen-dir=/var/fluxd/keygen",
+                  "--git-url="+giturl,
+                  "--git-branch=master"
+                ],
+                "image": "quay.io/weaveworks/flux:1.4.2",
+                "imagePullPolicy": "IfNotPresent",
+                "name": "flux",
+                "ports": [
+                  {
+                    "containerPort": 3030
+                  }
+                ],
+                "volumeMounts": [
+                  {
+                    "mountPath": "/etc/fluxd/ssh",
+                    "name": "git-key",
+                    "readOnly": true
+                  },
+                  {
+                    "mountPath": "/var/fluxd/keygen",
+                    "name": "git-keygen"
+                  }
+                ]
+              }
+            ],
+            "serviceAccount": "flux",
+            "volumes": [
+              {
+                "name": "git-key",
+                "secret": {
+                  "defaultMode": 256,
+                  "secretName": "flux-git-deploy"
+                }
+              },
+              {
+                "emptyDir": {
+                  "medium": "Memory"
+                },
+                "name": "git-keygen"
+              }
+            ]
+          }
+        }
+      }
+    },
+
+    nodeport:: {
+      // K8s Deployment,
+      "apiVersion": "v1",
+      "kind": "Service",
+      "metadata": {
+        "name": "flux"
+      },
+      "spec": {
+        "ports": [
+          {
+            "port": 80,
+            "targetPort": 3030
+          }
+        ],
+        "selector": {
+          "name": "flux"
+        },
+        "type": "NodePort"
+      }
+    },
+
+    fluxlb:: {
+      // K8s Deployment,
+      "apiVersion": "v1",
+      "kind": "Service",
+      "metadata": {
+        "labels": {
+          "app": "flux"
+        },
+        "name": "flux-lb"
+      },
+      "spec": {
+        "externalTrafficPolicy": "Cluster",
+        "ports": [
+          {
+            "port": 3030,
+            "protocol": "TCP",
+            "targetPort": 3030
+          }
+        ],
+        "selector": {
+          "name": "flux"
+        },
+        "type": "LoadBalancer"
```

Comment:
This would expose a port by default that's open to the world. We don't want to do this. This should be a parameter and users should have to explicitly choose to set the type to LoadBalancer in order to open up a port.

Reply:
Suggestions on how to fix this? I am thinking of a 

ks param set <deployment> LoadBalancer Deploy

I am just not sure how the code would look in the parameters and all.libsonnet

Also, should we give them more than one option for say a proxy? 

ks param set <deployment> port (Proxy/LoadBalancer)?

Reply:
Looks like this still needs to be fixed.

Reply:
We need to expose the flux service to the outside world so that the "fluxctl" CLI tool can access the tool. The LoadBalancer was to prevent a user from having to do a portforward. I think a better option may be a proxy option as the service does need to be exposed